packages/ihe-gateway-sdk/src/common/shared.ts

```diff
@@ -0,0 +1,20 @@
+import BadRequestError from "@metriport/core/util/error/bad-request";
+
+export function normalizeOid(oid: string | undefined): string {
```

Comment:
You might want to compile the regex outside the function to avoid recompilation on the regex each time. Especially if this normalization is called over and over again. So like:

`// Compile regex outside the function to avoid re-compilation on every call
const oidRegex = /^[0-9]+(\.[0-9]+)*$/;
const urnOidRegex = /^urn:oid:([0-9]+(\.[0-9]+)*)$/;

export function normalizeOid(oid: string | undefined): string {
....
}
 `